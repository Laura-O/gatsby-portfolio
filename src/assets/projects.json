[
  {
    "id": 0,
    "name": "TodoTron 3000",
    "url": "https://github.com/Laura-O/todo-tron",
    "description": "Productivity timer for todo.txt",
    "techs": ["Electron", "Vue.js", "Vuex"],
    "image": "/projects/todotron.png",
    "thumbnail": "/projects/todotron_thumb.png"
  },
  {
    "id": 1,
    "name": "FriendZone",
    "url": "https://github.com/Laura-O/social-network",
    "description": "A social network with chatroom",
    "techs": ["React", "Redux", "PostgreSQL", "Socket.io"],
    "image": "/projects/social_network.png",
    "thumbnail": "/projects/social_network_thumb.png"
  },
  {
    "id": 2,
    "name": "Portfolio Website",
    "url": "http://www.laura.fyi",
    "description":
      "This is my portfolio website. Look around to see the features!",
    "techs": ["Vue.js", "firebase"],
    "image": "/projects/portfolio.png",
    "thumbnail": "/projects/portfolio/portfolio_thumb.png"
  },
  {
    "id": 3,
    "name": "Petition",
    "url": "https://peachtition.herokuapp.com/",
    "description":
      "This is a website for a petition. Users can register, add information to their profile and sign the petition.",
    "techs": ["Express", "PostgreSQL", "Redis", "Handlebars", "JQuery"],
    "image": "/projects/landing_page.png",
    "thumbnail": "/projects/petition/landing_page_thumb.png"
  },
  {
    "id": 4,
    "name": "Connect 4",
    "url": "https://connect-4-game.herokuapp.com/",
    "description":
      "This is an implementation of the classic Connect-4 game for 2 players.",
    "techs": ["JQuery", "CSS", "HTML"],
    "image": "/projects/connect-4.png",
    "thumbnail": "/projects/connect-4/connect-4_thumb.png"
  },
  {
    "id": 5,
    "name": "Mobile flashcards",
    "url": "",
    "description":
      "A simple flashcard App built with React Native. You can create decks with questions, add questions and take a quiz. There is also a function to create daily reminders.",
    "techs": ["React Native", "React Redux"],
    "image": "/projects/quiz_page.png",
    "thumbnail": "/projects/mobile_flashcards/quiz_page_thumb.png"
  },
  {
    "id": 6,
    "name": "Bubble diagram D3.js",
    "url": "http://secure-meadow-44103.herokuapp.com/",
    "description":
      "This is an interactive bubble diagram I created from a database export of the World Cube Association WCA. It visualizes the competitors who attended the most competitions.",
    "techs": ["D3.js", "R", "Bootstrap"],
    "image": "/projects/d3-stats.png",
    "thumbnail": "/projects/d3_stats/d3-stats_thumb.png"
  }
]
