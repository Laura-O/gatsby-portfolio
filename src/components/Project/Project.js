import React, { Fragment } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Tag from '../../layouts/shared/tag';
import * as animations from '../../layouts/shared/animations';
import * as constants from '../../layouts/shared/style-constants';

const Techtag = styled(Tag)`
  padding: 4px 10px;
  margin: 5px;
`;

const ProjectWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  padding: 1.5rem;
  color: ${constants.textColor};
  width: 90%;
`;

const Left = styled.div`
  display: flex;
  flex: 1;
  animation: ${animations.moveInLeft} 1s ease-out;
  justify-content: center;
  align-items: center;
`;

const Right = styled.div`
  flex: 2;
  /* flex-wrap: wrap; */
  animation: ${animations.moveInRight} 1s ease-out;

  display: flex;
  flex-direction: column;
  justify-content: space-between;

  .list {
    margin-top: auto;
  }
`;

const Project = ({ project }) => {
  const listItems = project.techs.map(tech => (
    <Techtag key={tech}>{tech}</Techtag>
  ));
  return (

    <ProjectWrapper>
      <Left>
        <img src={project.thumbnail} alt={project.name} />
      </Left>
      <Right>
        <div><h3>
          <a href={project.url}>{project.name}</a>
        </h3></div>
        <div>{project.description}</div>
        <div className="list"><ul>{listItems}</ul></div>
      </Right>
    </ProjectWrapper>

  );
};

export default Project;

Project.propTypes = {
  project: PropTypes.shape({
    name: PropTypes.string.isRequired,
    description: PropTypes.string,
  }),
};
