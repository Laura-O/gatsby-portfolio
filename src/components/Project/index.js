import React from 'react'
import Link from 'gatsby-link'

import styles from './styles.module.scss'

const Project = ({ project }) => {
  const listItems = project.techs.map(tech => <li>{tech}</li>)

  return (
    <div className={styles.project}>
      <div className={styles.left}>
        <img src={project.thumbnail} />
      </div>
      <div className={styles.right}>
        <h3>{project.name}</h3>
        <div>{project.description}</div>
        <ul>{listItems}</ul>
      </div>
    </div>
  )
}

export default Project
