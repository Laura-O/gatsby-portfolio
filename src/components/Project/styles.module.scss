@import '../../scss/variables.scss';

.project {
  display: flex;
  flex-direction: row;
  padding: 1.5rem;
  align-items: center;
  justify-content: center;
}

.left {
  flex: 0 1 50%;
  align-self: center;
  animation: moveInLeft 1s ease-out;

  img {
    display: block;
    margin-left: auto;
    margin-right: auto;
  }
}

.description {
  max-width: 400px;
}

.right {
  flex: 0 1 50%;
  flex-wrap: wrap;
  align-items: flex-start;
  animation: moveInRight 1s ease-out;
}

.tag {
  border: 1px solid #dee2e5;
  background-color: #dee2e5;
  border-radius: 5px;
  padding: 6px 15px;
  margin: 5px;
  color: #000;
  transition: all 300ms ease-in-out;
  display: inline-block;

  &:hover {
    transform: translateY(-0.1rem);
    box-shadow: 0 1rem 2rem rgba(#000, 0.2);
    color: #fff;
    background-color: $linkColor;
  }

  &::after {
    transform: scaleX(1.4) scaleY(1.6);
    opacity: 0;
  }
}

@keyframes moveInLeft {
  0% {
    opacity: 0;
    transform: translateX(-10rem);
  }
  80% {
    transform: translate(1rem);
  }
  100% {
    opacity: 1;
    transform: translate(0);
  }
}

@keyframes moveInRight {
  0% {
    opacity: 0;
    transform: translateX(10rem);
  }
  80% {
    transform: translate(-1rem);
  }
  100% {
    opacity: 1;
    transform: translate(0);
  }
}
