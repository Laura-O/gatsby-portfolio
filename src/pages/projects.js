import React from 'react';
import styled from 'styled-components';
import Footer from '../components/Footer/Footer';
import Project from '../components/Project/Project';
import projects from '../assets/projects.json';

const ProjectPage = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  background-color: white;
  width: 100%;
  color: black;
  align-items: center;
  padding-top: 5em;
`;

const Divider = styled.div`
  position: relative;
  margin: 2rem;
  height: 0.1rem;

  &:before {
    content: '';
    position: absolute;
    left: 5%;
    right: 5%;
    width: 90%;
    height: 2px;
    box-shadow: -1px 0 0 0 #2ec4b6;
    background-image: linear-gradient(
      to right,
      transparent,
      #2ec4b6,
      transparent
    );
  }
`;

const ProjectsPage = () => (
  <div>
    <ProjectPage>
      <h1>Projects</h1>
      {projects.map(project => (
        <div key={project.id}>
          <Project project={project} key={project.id} />
          <Divider />
        </div>
      ))}
      <Footer />
    </ProjectPage>
  </div>
);

export default ProjectsPage;
