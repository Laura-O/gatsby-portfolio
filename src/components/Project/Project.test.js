import React from 'react';
import { shallow } from 'enzyme';
import Project from './Project';

import projects from '../../assets/projects.json';

const first = projects[0];

test('rendering', () => {
  const wrapper = shallow(<Project project={first} />);
  expect(wrapper).toMatchSnapshot();
});
