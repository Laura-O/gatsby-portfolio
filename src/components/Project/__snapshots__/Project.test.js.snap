// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering 1`] = `
<div>
  <div>
    <img
      alt="TodoTron 3000"
      src="/projects/todotron_thumb.png"
    />
  </div>
  <div>
    <h3>
      TodoTron 3000
    </h3>
    <div>
      Productivity timer for todo.txt
    </div>
    <ul>
      <span
        key="Electron"
      >
        Electron
      </span>
      <span
        key="Vue.js"
      >
        Vue.js
      </span>
      <span
        key="Vuex"
      >
        Vuex
      </span>
    </ul>
  </div>
</div>
`;
